test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,982 @@
+						collectedBundleVersions = append(collectedBundleVersions, pkgValue.Version)
+					}
+				}
+			}
+		}
+	}
+	if !reflect.DeepEqual(collectedBundleVersions, versions) {
+		return fmt.Errorf("Package %v for the catalog %v is not created", pkgName, operatorCatalog.Name)
+	}
+
+	return nil
+}
+
+// Checks for a successful resolution and bundle path for the operator
+func checkResolutionAndBundlePath(operator *operatorv1alpha1.Operator) error {
+	err := c.Get(ctx, types.NamespacedName{Name: operator.Name}, operator)
+	if err != nil {
+		return fmt.Errorf("Error retrieving operator %v:%v", operator.Name, err)
+	}
+	cond := apimeta.FindStatusCondition(operator.Status.Conditions, operatorv1alpha1.TypeResolved)
+	expectedCond := &metav1.Condition{
+		Type:    operatorv1alpha1.TypeResolved,
+		Status:  metav1.ConditionTrue,
+		Reason:  operatorv1alpha1.ReasonSuccess,
+		Message: "resolved to",
+	}
+	err = checkConditionEquals(cond, expectedCond)
+	if err != nil {
+		return fmt.Errorf("Status conditions for the operator %v for the version %v is not as expected:%v", operator.Name, operator.Spec.Version, err)
+	}
+	if operator.Status.ResolvedBundleResource == "" {
+		return fmt.Errorf("Resoved Bundle Resource is not found for the operator %v for the version %v", operator.Name, operator.Spec.Version)
+	}
+	return nil
+}
+
+// Checks if the operator installation succeeded
+func checkOperatorInstalled(operator *operatorv1alpha1.Operator, operatorVersion string) error {
+	err := c.Get(ctx, types.NamespacedName{Name: operator.Name}, operator)
+	if err != nil {
+		return fmt.Errorf("Error retrieving operator %v:%v", operator.Name, err)
+	}
+	cond := apimeta.FindStatusCondition(operator.Status.Conditions, operatorv1alpha1.TypeInstalled)
+	expectedCond := &metav1.Condition{
+		Type:    operatorv1alpha1.TypeResolved,
+		Status:  metav1.ConditionTrue,
+		Reason:  operatorv1alpha1.ReasonSuccess,
+		Message: "installed from",
+	}
+	err = checkConditionEquals(cond, expectedCond)
+	if err != nil {
+		return fmt.Errorf("Status conditions for the operator %v for the version %v is not as expected:%v", operator.Name, operator.Spec.Version, err)
+	}
+	if operator.Status.InstalledBundleResource == "" {
+		return fmt.Errorf("Installed Bundle Resource is not found for the operator %v for the version %v", operator.Name, operator.Spec.Version)
+	}
+	if operator.Spec.Version != operatorVersion {
+		return fmt.Errorf("Expected operator version: %s for the operator %v, but got: %s", operator.Spec.Version, operator.Name, operatorVersion)
+	}
+	return nil
+}
+
+// Checks if bundle deployment succeeded
+func checkPackageInstalled(operator *operatorv1alpha1.Operator) error {
+	bd := rukpakv1alpha1.BundleDeployment{}
+	err := c.Get(ctx, types.NamespacedName{Name: operator.Name}, &bd)
+	if err != nil {
+		return fmt.Errorf("Error retrieving the bundle deployments for the operator %v:%v", operator.Name, err)
+	}
+	if len(bd.Status.Conditions) != 2 {
```

Comment:
you might want to specifically look for conditions with `api.FindStatusCondition` here